vic/drivers/classic/src/alloc_veg_hist.c

```diff
@@ -38,29 +38,27 @@ alloc_veg_hist(int                nrecs,
 {
     int i, j;
 
-    (*veg_hist) = (veg_hist_struct **) calloc(nrecs, sizeof(veg_hist_struct *));
+    (*veg_hist) = calloc(nrecs, sizeof(*(*veg_hist)));
     if ((*veg_hist) == NULL) {
         log_err("Memory allocation error in alloc_veg_hist().");
     }
 
     for (i = 0; i < nrecs; i++) {
-        (*veg_hist)[i] = (veg_hist_struct *) calloc(nveg,
-                                                    sizeof(veg_hist_struct));
+        (*veg_hist)[i] = calloc(nveg, sizeof(*(*veg_hist)[i]));
```

Comment:
For these cases I would maintain parentheses, because it is difficult to remember whether `[]` has precedence over `*`.  However, on the right-hand-side I would place the parentheses differently. I should be `*(quantity on the left)` so:

`sizeof(*((*veg_hist)[i]))` which is admittedly ugly - I'll have to think about this.

However, since I think the `*` has precedence, I am not sure that 

`sizeof(*(*veg_hist)[i])` is correct (I wonder whether this is read as the ith element of **veg_hist, which is not what you want.

Same for many below. I'll look this up as well.